pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -111,12 +114,34 @@ public static boolean containsFunction(String functionName) {
    * methods are already registered.
    */
   @Nullable
-  public static FunctionInfo getFunctionInfo(String functionName, int numParameters) {
-    Map<Integer, FunctionInfo> functionInfoMap = FUNCTION_INFO_MAP.get(canonicalize(functionName));
-    return functionInfoMap != null ? functionInfoMap.get(numParameters) : null;
+  public static FunctionInfo getFunctionInfo(String functionName, String[] paramClassName) {
+    String paramTypeString = getSortedParamClassNameString(paramClassName);
+    Map<String, FunctionInfo> functionInfoMap = FUNCTION_INFO_MAP.get(canonicalize(functionName));
+    return functionInfoMap != null ? functionInfoMap.get(paramTypeString) : null;
   }
 
   private static String canonicalize(String functionName) {
     return StringUtils.remove(functionName, '_').toLowerCase();
   }
+
+  private static String[] getParamClassNameArray(Method method) {
+    Class[] paramClass = method.getParameterTypes();
+    String[] paramClassName = new String[paramClass.length];
+    for (int i = 0; i < paramClass.length; i++) {
+      paramClassName[i] = paramClass[i].getSimpleName();
+    }
+    return paramClassName;
+  }
+
+  public static String getSortedParamClassNameString(String[] paramClassName) {
+    Arrays.sort(paramClassName);
```

Comment:
Why do we sort the parameters? `func(int a, long b)` is not the same as `func(long a, int b)`

Reply:
I feel it's more complicated, and we shall consider some implicit type casting rule, especially for the numeric type.

For example, divide(5L, 5) shall match divide(long, long)  but not divide(int, int). i.e. if any of the argument is of higher-precision type, we shall pick the corresponding function not to lose precision.